.github/workflows/poetry-update.yaml

```diff
@@ -0,0 +1,19 @@
+name: Update Python Poetry packages
+
+on:
+  # Allow manual triggers.
+  workflow_dispatch:
+
+  # Automatically run once a week.
+  schedule:
```

Comment:
Should probably have another one that relocks on push to main when the individual pyproject files change, so we can always have an up-to-date lock file as well